### ISSUE TITLE
iOS app document & data storage size will grow very large over time when not relaunched.

### ISSUE THREAD

**Issue by jaymunro** [home-assistant/iOS#2275]
**iOS device model, version and app version**

Model Name: iPad (various)
Software Version: iOS 12.5.6
App version: 2022.10.1

**Home Assistant Core Version**
2022.12.8

**Describe the bug**
This is as per #2207 which was closed after #2231 was implemented.

However this is still an issue with “kiosk” like devices that just have the app running 24/7 and never re-launch.

Can a periodic compaction, say every 24 hours or so, be kicked off to resolve this?

**To Reproduce**
Leave the iOS app running in the foreground for a week or so.

**Expected behavior**
Realm database does not grow to gigabytes in size and consume all available disk space.

**Screenshots**

**Additional context**
Re-launching the app does indeed compact the db, but this should not be an ongoing requirement for families that have an iPad mounted on the wall dedicated to home control.

**Comment by jaymunro** [comment]
An update.
It seems that around 10% of a 16GB iPad is being consumed every 24 hours. This shows that a compaction at least every 24 hours is needed, if not more often.

![B27B59FD-5046-404B-AD97-DE266F6459F1](https://user-images.githubusercontent.com/1166024/210662816-285a6df6-55fc-43cb-a451-ac99d3e20dc6.png)

**Comment by jaymunro** [comment]
The issue appears to have gotten worse, with 6.7 GB being consumed in 10 hours, was about 3 days previously.
![IMG_0002](https://user-images.githubusercontent.com/1166024/213055146-3f3d8358-0969-4c6b-acd6-21f436dd3ca8.PNG)

The storaged used graph below shows a change from about 10 pm on Jan 10.
![IMG_0099](https://user-images.githubusercontent.com/1166024/213055180-3c9e0151-ea55-4b49-920f-e488e46f01b5.PNG)

I am only aware of an update from 2023.1.0 to 2023.1.1 at about 6:30 pm on Jan 9, so no correlation there.

**Comment by zacwest** [comment]
Realm 10.35.0 gained the ability to live compact, which should help this. https://github.com/realm/realm-swift/releases/tag/v10.35.0

**Comment by jaymunro** [comment]
Great.
Does this need to be bumped to be included in an update of the iOS app for HA?

**Comment by zacwest** [comment]
Yep, this will require an app release and upgrading the dependency

**Comment by zacwest** [comment]
Please try the latest [TestFlight build](https://home-assistant.io/ios/beta) and see if it resolves the issue for you.

**Comment by jaymunro** [comment]
Installed for testing now. Will keep this thread updated.

**Comment by jaymunro** [comment]
The TestFlight beta app seems to have made no difference.
![169B6FD9-BB4F-4978-828F-F50C9EDD1BFF](https://user-images.githubusercontent.com/1166024/220317636-cf59d59a-07c7-434a-bb7d-1f8018ea1a92.png)

**Comment by jaymunro** [comment]
Since installing the test flight beta on Feb 19, the app doesn’t even compact after restarts. Have even tried force quitting the app (flick up in running app list) but no compacting.

![8FC652E1-0C4A-4097-B1F0-C9B027CCFBDE](https://user-images.githubusercontent.com/1166024/221451213-051d8455-4201-4d97-8a3d-195964b695ec.png)
